mage_ai/data_cleaner/transformer_actions/column.py

```diff
@@ -196,6 +198,34 @@ def remove_column(df, action, **kwargs):
     return df.drop(columns=drop_columns)
 
 
+def remove_outliers(df, action, **kwargs):
+    cols = set(action['action_arguments'])
+    numeric_df = df.copy()
+    for column in df.columns:
+        if column not in cols:
+            numeric_df.drop(column, axis=1, inplace=True)
+        else:
+            dtype = action['action_variables'][column]['feature']['column_type']
+            if dtype in NUMBER_TYPES:
+                numeric_df.loc[:, column] = numeric_df.loc[:, column].astype(float)
+            else:
+                numeric_df.drop(column, axis=1, inplace=True)
+    outlier_mask = numeric_df.notna().all(axis=1)
+    numeric_df = numeric_df.dropna(axis=0)
```

Comment:
Another option is to fillna with 0. Do you have any particular reason of choosing to dropna here?

Reply:
The general problem with rows that have na values is that it is hard to infer whether they are outliers or not due to the missing information. If we fill the na values with zero and then identify that row as an outlier, it is possible that the row actually isn't an outlier and has been incorrectly removed. For most distributions which are likely not zero centered, these rows with NA values will likely get marked as outliers and removed despite there being no actual evidence to suggest that they are.

I thought that erring on the side of caution and not removing an outlier is better than accidentally removing a row that isn't an outlier, so I opted to drop the null valued rows instead.

Reply:
got it. that makes sense. the downside of dropna is if there're columns with mostly na, we'll drop the most of the rows. we can optimize for that case later.